### DISCUSSION TITLE
Store generated files into database

### DISCUSSION THREAD

**Discussion by inix** [facebook/docusaurus#4287] [General] [answered]
I want to store generated files into database so that I can auth which user can visit pages,but I find that some js files' name are generated differently each time I run build cmd. 
Can anyone help? 
Thanks

**Comment by slorber** [comment, marked as answer]
Not clear to me what you are really trying to achieve here.

Docusaurus is Jamstack so it output static HTML and js files. The JS file names should not matter and you should not rely on them to protect your site.

If you need to have an auth section on your site, you can instead create pages that are "browser only", and will only display data after your API has validated that the user is authenticated.

https://v2.docusaurus.io/docs/docusaurus-core#browseronly

```jsx
export default AuthenticatedPage() {
return (
    <BrowserOnly fallback={<Spinner/>}>
      {() => (
       <AuthenticatedContent/>
      )}
    </BrowserOnly>
  );
}
```

AuthenticatedContent is a component you have to build that will call your API, check for auth, return appropriate sensitive/user-specific data to render on the page, and then render that data on the client.

**Reply by inix** [reply]
> Not clear to me what you are really trying to achieve here.
> 
> Docusaurus is Jamstack so it output static HTML and js files. The JS file names should not matter and you should not rely on them to protect your site.
> 
> If you need to have an auth section on your site, you can instead create pages that are "browser only", and will only display data after your API has validated that the user is authenticated.
> 
> https://v2.docusaurus.io/docs/docusaurus-core#browseronly
> 
> ```js
> export default AuthenticatedPage() {
> return (
>     <BrowserOnly fallback={<Spinner/>}>
>       {() => (
>        <AuthenticatedContent/>
>       )}
>     </BrowserOnly>
>   );
> }
> ```
> 
> AuthenticatedContent is a component you have to build that will call your API, check for auth, return appropriate sensitive/user-specific data to render on the page, and then render that data on the client.

Hi, @slorber
Thanks for your reply.
I have my own PHP backend service now. I want to write docs with docusaurus markdown feature and then copy css,js files to www root, I also need to store html docs content body into database, but docs' html header are rendered by PHP dynamically and auth by PHP backend.
I have learned that js and css files' name maybe different each time I run yarn build cmd and those html file header will change some lines such as from` "<link rel="preload" href="/runtime~main.d477546d.js" as="script">"` to `"<link rel="preload" href="/runtime~main.c4175d4c.js" as="script">"`.
Is there any tech like APIs that I can detect, copy or inject to my PHP rendered files?
Thanks

**Reply by slorber** [reply]
Sorry but it's hard for me to understand what you are trying to do, it looks very complicated to me. for little value, and I can't provide support for that.
If you want to fully authenticate a docusaurus app, you'd rather simple put it under a baseurl like `/my-docusaurus-site` and protect the whole baseurl, instead of trying to protect files one at a time based on filenames.

Docusaurus does not and will never guarantee stable name for the output js files, as they are generated by webpack, and the names changes according to the webpack version and the code-splitting heuristics configuration.

**Reply by inix** [reply]
Ok, I see. Thank you
